packages/core/src/Synchronizer.ts

```diff
@@ -307,7 +307,10 @@ export class Synchronizer extends EventEmitter {
      * block we'll poll many times quickly
      */
     async start() {
-        await this.setup()
+        for (;;) {
```

Comment:
I have a different approach [here](https://github.com/Unirep/auth/blob/main/src/Synchronizer.js#L104). I'd rather use a shared promise than poll on this. The same promise is used in many places, [here](https://github.com/Unirep/auth/blob/main/src/Synchronizer.js#L149), [here](https://github.com/Unirep/auth/blob/main/src/Synchronizer.js#L170). So we can check for setup before building trees or doing anything that needs the `settings` object